test/cookie.test.js

```diff
@@ -847,3 +847,39 @@ test('should not decorate fastify, request and reply if no secret was provided',
     message: 'req.unsignCookie is not a function'
   })
 })
+
+// addHookOnRequest: false
+test('disable parses incoming cookies', (t) => {
+  t.plan(7)
+  const fastify = Fastify()
+  fastify.register(plugin, { addHookOnRequest: false })
+
+  for (const hook of ['preValidation', 'preHandler']) {
+    fastify.addHook(hook, (req, reply, done) => {
+      t.notOk(req.cookies)
```

Comment:
what does this mean? can you make a more explicit check? like compare with undefined or null?